core/controllers/android_e2e_config.py

```diff
@@ -0,0 +1,301 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controller for initializing android specific structures."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from constants import constants
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import fs_services
+from core.domain import opportunity_services
+from core.domain import question_domain
+from core.domain import question_services
+from core.domain import rights_manager
+from core.domain import skill_domain
+from core.domain import skill_services
+from core.domain import state_domain
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.domain import user_services
+import feconf
+import python_utils
+
+
+class InitializeAndroidTestDataHandler(base.BaseHandler):
+    """Handler to initialize android specific structures."""
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {'POST': {}}
+
+    @acl_decorators.open_access
+    def post(self):
+        """This method generates the following structures:
+            Topic: A topic with both a test story and a subtopic.
+            Story: A story with 'android_interactions' as a exploration
+                node.
+            Exploration: 'android_interactions' from the local assets.
+            Subtopic: A dummy subtopic to validate the topic.
+            Skill: A dummy skill to validate the subtopic.
+
+            All these structures are used for Android end-to-end testing, the
+            purpose is to test the network requests in android by fetching the
+            topic from the dev server.
+
+            This handler can only be used in the dev server, It gives a
+            500 error in the production server.
+
+            Returns a empty json - {}, when the request is successful.
+
+            This is only called once and raises a exception if the topic is
+            already created or published.
+
+        Raises:
+            Exception. When used in production mode.
+            Exception. The topic is already created but not published.
+            Exception. The topic is already published.
+        """
+
+        if not constants.DEV_MODE:
+            raise Exception('Cannot load new structures data in production.')
```

Comment:
This will raise 500 error in production, we don't expect to raise 500 error intentionally*

Reply:
I've added your suggestion for the route.
Now it is expected to give a 404 in production, yet I think this is necessary here considering if the route is registered in dev_mode.

if you meant someother error can you please specify.

Reply:
I still see this raise Exception here* 

I think this will raise a 500 HTTP error code response which is not expected, it would raise 404 instead, right?

Reply:
I understand that calling this request should give a 404 error.
but for the 404 we need not mention it as an exception.

I'm leaving this exception, considering that if the route is registered in the production server (as we've not tested it in the production server yet). For some security reasons as this request involves admin as the COMMITTER_ID.

Let me know if this isn't required.